src/app/Http/Controllers/Operations/FetchOperation.php

```diff
@@ -70,11 +70,24 @@ private function fetch($arg)
             $config['query']->get();
         }
 
-        $textColumnTypes = ['string', 'json_string', 'text'];
+        // FetchOperation has an optional parameter in url that when present means we want to fetch a single entity.
+        $fetchUriCount = count(explode('/', request()->route()->uri())) - 1;
+        $currentUriCount = count(explode('/', request()->path()));
+
+        //in this case last parameter in url was specified so we will trigger the return of single entity
+        if ($currentUriCount > $fetchUriCount) {
+            return $model_instance->where($model_instance->getKeyName(), $search_string)->first();
```

Comment:
Oh I see now. We're using the ```$search_string``` to pass the ID from select2 to the Fetch operation. I understand now. 

Any reason why we don't use the new parameter to pass the exact ID? Then we could just check if the parameter exists - and if so, it's the ID.

Reply:
I did it this way because I am not 100% sure how it will work for multiple selects (I think the same) but the query string would be a json string representation of an array of id's.

That's why i used the parameter, as you said correctly, as a boolean switch.

We could use the parameter as ID, but we only register one post route and would endup POSTING empty form to emulate a get request.

This way seemed more intuitive to me. For multiple i think i just have to try to `json_decode` the `search_string` if we have an array `return $model->whereIn('key', [array, of, ids])`

Reply:
Oh I understand. I agree with you then, it would be cleaner to use POST variables when send multiple IDs. It would work with GET too, but it'll be messy, especially if the IDs are actually UUIDs or something.

Then the best way to do this might be to:
- remove this new parameter entirely
- in the AJAX request, in addition to using the "q" parameter, we could also send a "keys" parameter (as POST variables, not GET)
- then inside the Fetch operation, we can just check to see if ```request()->has('keys')``` or ```count(request('keys'))```, and if so make sure to apply the proper logic

I think this would:
- leave the route simple and pretty, no ```/single``` flag; it would also require no parameter counting logic or anything;
- use POST instead of GET, to account for sending multiple IDs;

What do you think @pxpm ? Sounds good?